### PR TITLE
Cleanup BFS7.7.1

### PR DESCRIPTION
**Pull request by 2samueld**
# Description:

This PR accomplishes the following: 
- Cleanup previous test logs on bfs7.7.1
- Create a temp optimize directory when running CiGroups in parallel (fixes optimization issue)
- Cleanup for what's been added/removed for testing purposes

### CODE REVIEW

**Review by Tengda-He — APPROVED** [review]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -93,10 +93,12 @@ node('test') {
 
 def functionalDynamicParallelSteps(image){
     ciGroupsMap = [:]
-    for (int i = 1; i <= 3; i++) {
+    for (int i = 1; i <= 12; i++) {
         def currentCiGroup = "ciGroup${i}";
         def currentStep = i;
         ciGroupsMap["${currentCiGroup}"] = {
+            sh "rm -rf ${env.WORKSPACE}_${currentCiGroup}"
+            sh "mkdir ${env.WORKSPACE}_${currentCiGroup}"
```
Comment (sichend) [inline Jenkinsfile:101]: This is not a good design. Please take a look at the last part of my comment. https://github.com/Tengda-He/kibana/issues/57#issuecomment-982808830
We should not extent the execution to the outside of the workspace. Please fix this

**Review by sichend — COMMENTED** [review]
I dont see the issue referenced to this PR. Please make sure reference the issue in the PR description. Please refer to this documentation. https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls#issues-and-pull-requests

**Review by sichend — COMMENTED** [review]
A couple of extra things to point out for your future reference. You can leave the current PR as is.
1. One commit should accomplish only one thing. If you are cleaning up, just clean up. Open a separate PR for the Jenkins file update. The logic is every commit in mainline is roll-backable.
2. Follow this guide line for your commit messages, https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/ Details are not required, but the message headline need to comply with the best practice.

File: test/common/config.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ export default function () {
       sourceArgs: [
         '--no-base-path',
         '--env.name=development',
+        `--optimize.bundleDir=${OPTIMIZE_BUNDLE_DIR}`,
```
[the remark this answers is not in the document]
Comment (2samueld) [reply test/common/config.js:45]: This was removed, when testing optimization issue.

File: Jenkinsfile [hunk 3]
```diff
@@ -111,91 +113,11 @@ def functionalDynamicParallelSteps(image){
...
-                    "TEST_KIBANA_HOST=localhost",
-                    "TEST_KIBANA_PORT=6610",
-                    "TEST_ES_TRANSPORT_PORT=9403",
-                    "TEST_ES_PORT=9400",
-                    "CI_PARALLEL_PROCESS_NUMBER=${currentStep}",
-                    "JOB=ci${currentStep}",
-                    "CACHE_DIR=${currentCiGroup}"
-                ]) {
-                    image.inside {
+                    image.inside("-v \'${env.WORKSPACE}_${currentCiGroup}:${env.WORKSPACE}/optimize\'") {
```
Comment (zhyuanqi) [inline Jenkinsfile:116]: Let's see if you can use the last approach. By creating multiple empty directories in the workspace each name after corresponding ciGroups. Then mount each directory to corresponding CI group while running functional test in Jenkins.

**Pull request merged**